sourcegraph/sourcegraph
# Code Insights: Remove insights api instance checks

Closes part of https://github.com/sourcegraph/sourcegraph/issues/31412

## Background
Historically we've been dealing with two different types of API in our UI. Setting-based API (currently deprecated) and GQL real API. We unified most of the methods of APIs when we were implementing the GQL API adoption but we left a few places in our UI component where we didn't extract the  Setting-based API in a generic API class. This PR removes these places. 

## Test plan
- Check that the creation UI navigation works properly (`dashboardId` URL query param should still redirect users to the right dashboard after creation) 
- Check that insight creation/editing works properly for all types of insights
